python_modules/libraries/dagster-snowflake-pandas/dagster_snowflake_pandas/snowflake_pandas_type_handler.py

```diff
@@ -1,8 +1,11 @@
 from typing import Mapping, Union, cast
 
 import pandas as pd
-from dagster import InputContext, MetadataValue, OutputContext, TableColumn, TableSchema
-from dagster._core.definitions.metadata import RawMetadataValue
+import pandas.core.dtypes.common as pd_core_dtypes_common
+from dagster._core.definitions.metadata import MetadataValue, RawMetadataValue
```

Comment:
These imports should come from `dagster`, no?

Reply:
Changed it but in general IMO no-- we don't enforce this and if you are working with a pyright/pylance instance that has both libs and dagster core in the workspace, then auto-import will resolve to the defining module rather than `dagster`.

Reply:
I think that integration libraries should lean towards using Dagster public APIs. This will allow us to eventually decouple the versions.